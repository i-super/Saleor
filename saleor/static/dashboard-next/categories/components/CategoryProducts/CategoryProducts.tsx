import AddIcon from "@material-ui/icons/Add";
import Cached from "@material-ui/icons/Cached";
import Avatar from "material-ui/Avatar";
import Card from "material-ui/Card";
import blue from "material-ui/colors/blue";
import IconButton from "material-ui/IconButton";
import { withStyles } from "material-ui/styles";
import Table, {
  TableBody,
  TableCell,
  TableFooter,
  TableHead,
  TableRow
} from "material-ui/Table";
import * as React from "react";

import PageHeader from "../../../components/PageHeader";
import Skeleton from "../../../components/Skeleton";
import TablePagination from "../../../components/TablePagination";
import i18n from "../../../i18n";

const decorate = withStyles(theme => ({
  avatarCell: {
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: 0,
    width: theme.spacing.unit * 5
  },
  link: {
    color: blue[500],
    cursor: "pointer"
  }
}));

interface ProductListProps {
  hasNextPage?: boolean;
  hasPreviousPage?: boolean;
  products?: Array<{
    id: string;
    name: string;
    productType: {
      name: string;
    };
    thumbnailUrl: string;
  }>;
  onAddProduct?();
  onNextPage?();
  onPreviousPage?();
  onRowClick?(id: string): () => void;
}

export const ProductList = decorate<ProductListProps>(
  ({
    classes,
    hasNextPage,
    hasPreviousPage,
    products,
    onAddProduct,
    onNextPage,
    onPreviousPage,
    onRowClick
  }) => (
    <Card>
      <PageHeader title={i18n.t("Products")}>
        {!!onAddProduct && (
          <IconButton onClick={onAddProduct}>
            <AddIcon />
          </IconButton>
        )}
      </PageHeader>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell className={classes.avatarCell} />
            <TableCell>{i18n.t("Name", { context: "object" })}</TableCell>
            <TableCell>{i18n.t("Type", { context: "object" })}</TableCell>
          </TableRow>
        </TableHead>
        <TableFooter>
          <TableRow>
            <TablePagination
              colSpan={3}
              hasNextPage={hasNextPage}
              onNextPage={onNextPage}
              hasPreviousPage={hasPreviousPage}
              onPreviousPage={onPreviousPage}
            />
          </TableRow>
        </TableFooter>
        <TableBody>
          {products === undefined || products === null ? (
            <TableRow>
              <TableCell className={classes.avatarCell}>
                <Avatar>
                  <Cached />
                </Avatar>
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
              <TableCell>
                <Skeleton />
              </TableCell>
            </TableRow>
          ) : products.length > 0 ? (
            products.map(product => (
              <TableRow key={product.id}>
                <TableCell className={classes.avatarCell}>
                  <Avatar src={product.thumbnailUrl} />
                </TableCell>
                <TableCell>
                  <span
                    onClick={onRowClick ? onRowClick(product.id) : () => {}}
                    className={onRowClick ? classes.link : ""}
                  >
                    {product.name}
                  </span>
                </TableCell>
                <TableCell>{product.productType.name}</TableCell>
              </TableRow>
            ))
          ) : (
            <TableRow>
              <TableCell className={classes.avatarCell} />
              <TableCell colSpan={2}>{i18n.t("No products found")}</TableCell>
            </TableRow>
          )}
        </TableBody>
      </Table>
    </Card>
  )
);

export default ProductList;
