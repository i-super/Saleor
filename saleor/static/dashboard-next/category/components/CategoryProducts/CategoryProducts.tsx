import AddIcon from "material-ui-icons/Add";
import FilterListIcon from "material-ui-icons/FilterList";
import Card from "material-ui/Card";
import Hidden from "material-ui/Hidden";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import PageHeader from "../../../components/PageHeader";
import ProductList from "../../../components/ProductList";
import i18n from "../../../i18n";

interface CategoryProductsProps {
  hasPreviousPage: boolean;
  hasNextPage: boolean;
  products?: Array<{
    id: string;
    name: string;
    thumbnailUrl: string;
    productType: {
      id: string;
      name: string;
    };
  }>;
  onCreate?();
  onFilter?();
  onNextPage?();
  onPreviousPage?();
}

const CategoryProducts: React.StatelessComponent<CategoryProductsProps> = ({
  hasNextPage,
  hasPreviousPage,
  onCreate,
  onFilter,
  onNextPage,
  onPreviousPage,
  products
}) => (
  <Card>
    <PageHeader
      title={i18n.t("Products", {
        context: "title"
      })}
    >
      <IconButton onClick={onCreate}>
        <AddIcon />
      </IconButton>
      <Hidden mdUp>
        <IconButton onClick={onFilter}>
          <FilterListIcon />
        </IconButton>
      </Hidden>
    </PageHeader>
    <ProductList
      hasNextPage={hasNextPage}
      hasPreviousPage={hasPreviousPage}
      onNextPage={onNextPage}
      onPreviousPage={onPreviousPage}
      products={products}
    />
  </Card>
);

export default CategoryProducts;
