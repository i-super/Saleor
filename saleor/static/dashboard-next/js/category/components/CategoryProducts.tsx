import AddIcon from "material-ui-icons/Add";
import FilterListIcon from "material-ui-icons/FilterList";
import Card from "material-ui/Card";
import Hidden from "material-ui/Hidden";
import IconButton from "material-ui/IconButton";
import * as React from "react";

import PageHeader from "../../components/PageHeader";
import ProductList from "../../components/ProductList";
import { CategoryPropertiesQuery } from "../../gql-types";
import i18n from "../../i18n";

interface CategoryProductsProps {
  data: CategoryPropertiesQuery;
  loading?: boolean;
  onCreate?();
  onFilter?();
  onNextPage?();
  onPreviousPage?();
}

const CategoryProducts: React.StatelessComponent<CategoryProductsProps> = ({
  data,
  loading,
  onCreate,
  onFilter,
  onNextPage,
  onPreviousPage
}) => (
  <Card>
    <PageHeader
      title={i18n.t("Products", {
        context: "title"
      })}
    >
      <IconButton disabled={loading} onClick={onCreate}>
        <AddIcon />
      </IconButton>
      <Hidden mdUp>
        <IconButton disabled={loading} onClick={onFilter}>
          <FilterListIcon />
        </IconButton>
      </Hidden>
    </PageHeader>
    <ProductList
      hasNextPage={
        data.category &&
        data.category.products &&
        data.category.products.pageInfo &&
        data.category.products.pageInfo.hasNextPage
      }
      hasPreviousPage={
        data.category &&
        data.category.products &&
        data.category.products.pageInfo &&
        data.category.products.pageInfo.hasPreviousPage
      }
      onNextPage={onNextPage}
      onPreviousPage={onPreviousPage}
      products={
        data.category &&
        data.category.products &&
        data.category.products.edges.map(edge => edge.node)
      }
    />
  </Card>
);

export default CategoryProducts;
