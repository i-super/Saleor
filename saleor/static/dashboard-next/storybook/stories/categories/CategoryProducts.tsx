import { storiesOf } from "@storybook/react";
import { DialogContentText } from "material-ui/Dialog";
import * as React from "react";

import CategoryProducts from "../../../category/components/CategoryProducts";

const products = [
  {
    id: "UHJvZHVjdDox",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoy",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoz",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoa",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  }
];

storiesOf("Categories / CategoryProducts", module)
  .add("with initial data", () => (
    <CategoryProducts
      hasNextPage={true}
      hasPreviousPage={false}
      products={products}
      onCreate={() => {}}
      onFilter={() => {}}
      onNextPage={() => {}}
      onPreviousPage={() => {}}
    />
  ))
  .add("without initial data", () => (
    <CategoryProducts
      hasNextPage={true}
      hasPreviousPage={false}
      products={[]}
      onCreate={() => {}}
      onFilter={() => {}}
      onNextPage={() => {}}
      onPreviousPage={() => {}}
    />
  ))
  .add("when loading data", () => (
    <CategoryProducts
      hasNextPage={true}
      hasPreviousPage={false}
      onCreate={() => {}}
      onFilter={() => {}}
      onNextPage={() => {}}
      onPreviousPage={() => {}}
    />
  ));
