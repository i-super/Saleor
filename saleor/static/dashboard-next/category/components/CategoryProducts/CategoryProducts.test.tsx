import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import CategoryProducts from "./CategoryProducts";

const products = [
  {
    id: "UHJvZHVjdDox",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoy",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoz",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoa",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  }
];

describe("<CategoryProducts />", () => {
  it("renders with initial data", () => {
    const component = renderer.create(
      <CategoryProducts
        hasNextPage={true}
        hasPreviousPage={false}
        products={products}
        onCreate={jest.fn()}
        onFilter={jest.fn()}
        onNextPage={jest.fn()}
        onPreviousPage={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders without initial data", () => {
    const component = renderer.create(
      <CategoryProducts
        hasNextPage={true}
        hasPreviousPage={false}
        products={[]}
        onCreate={jest.fn()}
        onFilter={jest.fn()}
        onNextPage={jest.fn()}
        onPreviousPage={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when loading data", () => {
    const component = renderer.create(
      <CategoryProducts
        hasNextPage={true}
        hasPreviousPage={false}
        onCreate={jest.fn()}
        onFilter={jest.fn()}
        onNextPage={jest.fn()}
        onPreviousPage={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
