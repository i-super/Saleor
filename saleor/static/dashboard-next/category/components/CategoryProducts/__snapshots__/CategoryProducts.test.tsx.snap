// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryProducts /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <h2
      className="MuiTypography-root-34 MuiTypography-title-40 Component-title-33"
    >
      Products
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-64 MuiIconButton-root-58"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-73"
        />
      </button>
    </div>
  </div>
  <table
    className="MuiTable-root-81"
  >
    <thead
      className="MuiTableHead-root-82"
    >
      <tr
        className="MuiTableRow-root-83 MuiTableRow-head-84"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89 Component-avatarCell-80"
          scope="col"
        />
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89"
          scope="col"
        >
          Type
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr
        className="MuiTableRow-root-83 MuiTableRow-footer-85"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-footer-91 MuiTablePagination-root-99"
          colSpan={3}
          scope={undefined}
        >
          <div
            className="MuiToolbar-root-105 MuiToolbar-gutters-106 MuiTablePagination-toolbar-104"
          >
            <div
              className="MuiTablePagination-spacer-103"
            />
            <div
              className="MuiTablePaginationActions-root-107"
            >
              <button
                className="MuiButtonBase-root-64 MuiButtonBase-disabled-65 MuiIconButton-root-58 MuiIconButton-disabled-62"
                disabled={true}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiIconButton-label-63"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-67"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                className="MuiButtonBase-root-64 MuiIconButton-root-58"
                disabled={false}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-63"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-67"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-73"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
    <tbody
      className="MuiTableBody-root-108"
    >
      <tr
        className="MuiTableRow-root-83"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90 Component-avatarCell-80"
          scope={undefined}
        >
          <div
            className="MuiAvatar-root-109 MuiAvatar-colorDefault-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 8l-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3l-4-4zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4H6z"
              />
            </svg>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          <span
            className="Component-skeleton-112"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          <span
            className="Component-skeleton-112"
            style={undefined}
          >
            ‌
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<CategoryProducts /> renders with initial data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <h2
      className="MuiTypography-root-34 MuiTypography-title-40 Component-title-33"
    >
      Products
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-64 MuiIconButton-root-58"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-73"
        />
      </button>
    </div>
  </div>
  <table
    className="MuiTable-root-81"
  >
    <thead
      className="MuiTableHead-root-82"
    >
      <tr
        className="MuiTableRow-root-83 MuiTableRow-head-84"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89 Component-avatarCell-80"
          scope="col"
        />
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89"
          scope="col"
        >
          Type
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr
        className="MuiTableRow-root-83 MuiTableRow-footer-85"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-footer-91 MuiTablePagination-root-99"
          colSpan={3}
          scope={undefined}
        >
          <div
            className="MuiToolbar-root-105 MuiToolbar-gutters-106 MuiTablePagination-toolbar-104"
          >
            <div
              className="MuiTablePagination-spacer-103"
            />
            <div
              className="MuiTablePaginationActions-root-107"
            >
              <button
                className="MuiButtonBase-root-64 MuiButtonBase-disabled-65 MuiIconButton-root-58 MuiIconButton-disabled-62"
                disabled={true}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiIconButton-label-63"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-67"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                className="MuiButtonBase-root-64 MuiIconButton-root-58"
                disabled={false}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-63"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-67"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-73"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
    <tbody
      className="MuiTableBody-root-108"
    >
      <tr
        className="MuiTableRow-root-83"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90 Component-avatarCell-80"
          scope={undefined}
        >
          <div
            className="MuiAvatar-root-109"
          >
            <img
              alt={undefined}
              className="MuiAvatar-img-111"
              sizes={undefined}
              src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg"
              srcSet={undefined}
            />
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          Gardner, Graham and King
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          T-Shirt
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-83"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90 Component-avatarCell-80"
          scope={undefined}
        >
          <div
            className="MuiAvatar-root-109"
          >
            <img
              alt={undefined}
              className="MuiAvatar-img-111"
              sizes={undefined}
              src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg"
              srcSet={undefined}
            />
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          Gardner, Graham and King
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          T-Shirt
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-83"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90 Component-avatarCell-80"
          scope={undefined}
        >
          <div
            className="MuiAvatar-root-109"
          >
            <img
              alt={undefined}
              className="MuiAvatar-img-111"
              sizes={undefined}
              src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg"
              srcSet={undefined}
            />
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          Gardner, Graham and King
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          T-Shirt
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-83"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90 Component-avatarCell-80"
          scope={undefined}
        >
          <div
            className="MuiAvatar-root-109"
          >
            <img
              alt={undefined}
              className="MuiAvatar-img-111"
              sizes={undefined}
              src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg"
              srcSet={undefined}
            />
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          Gardner, Graham and King
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          scope={undefined}
        >
          T-Shirt
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<CategoryProducts /> renders without initial data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <div
    className="Component-root-31"
  >
    <h2
      className="MuiTypography-root-34 MuiTypography-title-40 Component-title-33"
    >
      Products
    </h2>
    <div
      className="Component-action-28"
    >
      <button
        className="MuiButtonBase-root-64 MuiIconButton-root-58"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-63"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-67"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-73"
        />
      </button>
    </div>
  </div>
  <table
    className="MuiTable-root-81"
  >
    <thead
      className="MuiTableHead-root-82"
    >
      <tr
        className="MuiTableRow-root-83 MuiTableRow-head-84"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89 Component-avatarCell-80"
          scope="col"
        />
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-head-89"
          scope="col"
        >
          Type
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr
        className="MuiTableRow-root-83 MuiTableRow-footer-85"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-footer-91 MuiTablePagination-root-99"
          colSpan={3}
          scope={undefined}
        >
          <div
            className="MuiToolbar-root-105 MuiToolbar-gutters-106 MuiTablePagination-toolbar-104"
          >
            <div
              className="MuiTablePagination-spacer-103"
            />
            <div
              className="MuiTablePaginationActions-root-107"
            >
              <button
                className="MuiButtonBase-root-64 MuiButtonBase-disabled-65 MuiIconButton-root-58 MuiIconButton-disabled-62"
                disabled={true}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiIconButton-label-63"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-67"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                className="MuiButtonBase-root-64 MuiIconButton-root-58"
                disabled={false}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-63"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-67"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-73"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
    <tbody
      className="MuiTableBody-root-108"
    >
      <tr
        className="MuiTableRow-root-83"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90 Component-avatarCell-80"
          scope={undefined}
        />
        <td
          aria-sort={null}
          className="MuiTableCell-root-88 MuiTableCell-body-90"
          colSpan={2}
          scope={undefined}
        >
          No products found
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
